### ISSUE TITLE
[BUG] TBC Classic Item Icon not updating in bag

### ISSUE THREAD

**Issue by d3xavier** [arkayenro/arkinventory#1491]
**What version number you are using?**
ArkInventory-3.09.55-BurningCrusade

**What game client are you playing? Retail, Burning Crusade, Classic / Live, PTR, Beta**
TBC Classic

**What language is the game client set to?**
English

**Describe the bug**
When equipping an item from my bag, the icon remains the same as the item equipped, though the tooltips shows the item that was unequipped and placed into that bag slot.

**To Reproduce**
Steps to reproduce the behaviour, if possible:
equip an item from your bag.

**Screenshots**
![image](https://user-images.githubusercontent.com/4826700/129397386-76ed8d6c-4a63-4c9c-acab-e0a91ddce98a.png)
![image](https://user-images.githubusercontent.com/4826700/129397448-8a7cc0da-92cc-4019-ba76-1be0fc400cc9.png)

**Additional context**

**Comment by d3xavier** [comment]
In the second screenshot, I equipped Khadgar's Knapsack, you can see it on my character.  But the item icon remained (and greyed out) for the Knapsack, and the tooltip shows its the Jewel of Infinite Possibilities

**Comment by d3xavier** [comment]
I tried enabling Debug to see if there are any errors that I can capture, and I am not seeing any displayed.  Additionally, even when closing ArkInventory and re-opening, the bag slot still shows the previously present (but now equipped) item icon, but the tooltip shows the unequipped item currently occupying the bag slot.

**Comment by d3xavier** [comment]
ahh, I have found that this bug is present when first loading the game.  If I do a ReloadUI, the icons for the items do get updated on equip/unequip, but has a slight 150ms delay in the icon change.

**Comment by arkayenro** [comment]
could you try [r1145-alpha](https://www.curseforge.com/wow/addons/ark-inventory/files/3425036) and let me know if its fixed or not

**Comment by Gunner-web** [comment]
Hi, I am also seeing this issue but it happens in other areas as well.  I tried the alpha file listed above. I have localized the issue I see as being associated with items in the Backpack.  I can move items from a backpack slot to other bag slots and the item is grayed out in the backpack but shows normal.  See the Cheese wheel in the example.  I'm also not sure what the number in the top left corner of some icons is for.
![image](https://user-images.githubusercontent.com/88938346/129450799-6dda32d3-0def-4ff4-b2bb-128d9ad440d6.png)

**Comment by arkayenro** [comment]
actually could you try alpha 3 / [r1146-alpha](https://www.curseforge.com/wow/addons/ark-inventory/files/3425220) and let me know if its fixed or not

**Comment by arkayenro** [comment]
@Gunner-web the number in the top left is either the item level, or the stock count, if the item has those

**Comment by Gunner-web** [comment]
Just tried it and the issue remains.  this issue started showing up for me
after I updated from 3.09.52, if that helps any.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by d3xavier** [comment]
As Gunner mentioned, I tried upgrading as well to both r1145-alpha and then r1146-alpha as well.  issue remains, and I can verify if the item is not in the main Backpack and is instead in one of the other bags, the cache issue no longer seems to happen.

**Comment by arkayenro** [comment]
when you get a grey item can you enable edit mode and see if it goes away.
if not can you click on it and check the debug, is the data ready true or false?

**Comment by d3xavier** [comment]
![image](https://user-images.githubusercontent.com/4826700/129470142-47e9c907-b95e-44ee-99b3-668f1d800d44.png)

When I enable edit mode, the greyed out item remains.  Above is the screenshot of the debug of the item.  Data is Ready = True  
In this instance of the screenshot, the Frozen Shadoweave boots are equipped, bag slot shows the icon for Frozen Shadoweave Boots, and tooltip for the greyed out item is the Ruby Slippers.

**Comment by Nitti2909** [comment]
I have the same problem with icons not updating, in my case its new items in the bags that are not getting an icon at all, if i reload the game it works...
The new item in the bag is marked with a blue circle.
![Ark](https://user-images.githubusercontent.com/69142568/129475401-331d3dd9-2ec4-4e20-b649-1eab54e5183d.png)

**Comment by arkayenro** [comment]
try [alpha 6](https://www.curseforge.com/wow/addons/ark-inventory/files/3425982) and let me know if its any better

**Comment by Nitti2909** [comment]
Still the same issue with alpha 6...

**Comment by arkayenro** [comment]
make sure youre using alpha 7
click on switch character, mouse over the current character, select erase backpack data. see if that helps.

**Comment by Nitti2909** [comment]
After installing Alpha 7 and reset backpack data my issue didn't get fixed, it just got a bit weirder instead... 
I transferred 2 stacks of "Elixir of Ironskin" one with 15 and one with 20, between 2 characters they got registered as one stacks of 15 (Tooltip only registered me as having 15 in my bags). but you can see 2 stacks of 15 in my bags, when i then transferred them back to another character again they are registered as 1 stack of 15 and 1 stack of 20 as soon as they are outside Arkinventory...
Hope it helps in some way...
<img width="489" alt="Ark2" src="https://user-images.githubusercontent.com/69142568/129580616-bbc7c5b2-ed47-4d6f-908c-07963dd5f6da.png">

**Comment by arkayenro** [comment]
i found the issue, sort of.  i can duplicate it if i disable saving the backpack data, but theres nothing in the code that links that to the backpack not geting scanned (which is the actual issue)

**Comment by d3xavier** [comment]
I am using Alpha 7 now, and I tried to clear backpack data for my character.  I saw the message "ArkInventory: Saved backpack data for Demiz - Ashkandi has been erased".  ArkInventory closed, and when re-opening ArkInventory, the primary Backpack has "ERR" over it, and I can no longer see the items in the backpack.   I have to do a /reloadui to fix it.  After relogging from the game, the cacheing issue remains.

![image](https://user-images.githubusercontent.com/4826700/129642239-b4aed376-25d7-417e-9edc-30be729e055b.png)

**Comment by d3xavier** [comment]
This may be a stupid comment, or not relevant to the issue, but is it possible due to different backpack sizes?  I am wondering if someone who does not have 2fac authentication and has the normal Backpack size has a different experience than the enhanced Backpack given by Blizzard for having the security feature enabled for the account?

**Comment by arkayenro** [comment]
no, its a bug in old code that never, or only very rarely, ran until i added object caching, then it kicked in, mostly when you werent saving the backpack data, and would stop scanning the backpack/

should be fixed in [alpha 8](https://www.curseforge.com/wow/addons/ark-inventory/files/3428782)

**Comment by Nitti2909** [comment]
Mine is working fine now with Alpha 8...:-)))
Ones again m8, thx for a great addon, ive been using this from day one so about 15 years and i really appreciate all the work you put in to keeping it updated and working...:-)

**Comment by d3xavier** [comment]
amazing, great job @arkayenro.  caching issue is resolved, and the icon update (rescan?) seems to be faster now when equipping/unequipping items.  To echo was @Nitti2909 said, I have used this addon since the original TBC. and appreciate your time and effort all of these years.

**Comment by arkayenro** [comment]
check [alpha 10](https://www.curseforge.com/wow/addons/ark-inventory/files/3430371), that should fix the last of the issues, and stop any debug output

**Comment by Gunner-web** [comment]
Got this debug info after updating to alpha10
![image](https://user-images.githubusercontent.com/88938346/130084044-3939b046-25fe-44f5-bf38-dca7b68a9be3.png)
